### PR TITLE
Added acceptance tests for in process scenario.

### PR DESCRIPTION
**Pull request by Faizan2304**

### CODE REVIEW

File: test/Microsoft.TestPlatform.AcceptanceTests/Extension/CustomDataTestMethodAttribute.cs [hunk 1]
```diff
@@ -89,9 +104,58 @@ public class NETFullTargetFramework : Attribute
         /// </summary>
         public NETFullTargetFramework()
         {
-            this.DataRows = new List<DataRowAttribute>(2);
-            this.DataRows.Add(new DataRowAttribute(IntegrationTestBase.CoreRunnerFramework, AcceptanceTestBase.DesktopTargetFramework, AcceptanceTestBase.CoreRunnerTargetRuntime));
-            this.DataRows.Add(new DataRowAttribute(IntegrationTestBase.DesktopRunnerFramework, AcceptanceTestBase.DesktopTargetFramework, AcceptanceTestBase.DesktopRunnerTargetRuntime));
+            this.DataRows = new List<DataRowAttribute>(1);
```
Comment (smadala) [inline test/Microsoft.TestPlatform.AcceptanceTests/Extension/CustomDataTestMethodAttribute.cs]: Make this `NETFullTargetFrameworkWithCoreRunner`.

File: test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs [hunk 2]
```diff
@@ -92,15 +93,15 @@ public void RunMultipleTestAssembliesInParallel(string runnerFramework, string t
         }
 
         [CustomDataTestMethod]
-        [NETFullTargetFramework]
+        [NETFullTargetFramework(true, true)]
```
Comment (smadala) [inline test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs]: Readability: Change it to `[NETFullTargetFramework(inIsolation:true, inProcess:true)]`.

File: test/Microsoft.TestPlatform.AcceptanceTests/ResultsDirectoryTests.cs [hunk 3]
```diff
@@ -10,12 +10,12 @@ namespace Microsoft.TestPlatform.AcceptanceTests
     public class ResultsDirectoryTests : AcceptanceTestBase
     {
         [CustomDataTestMethod]
-        [NETFullTargetFramework]
+        [NETFullTargetFramework(true, true)]
```
Comment (smadala) [inline test/Microsoft.TestPlatform.AcceptanceTests/ResultsDirectoryTests.cs]: No need to run in `inprocess`.

File: test/Microsoft.TestPlatform.AcceptanceTests/ResultsDirectoryTests.cs [hunk 4]
```diff
@@ -31,12 +31,13 @@ public void TrxFileShouldBeCreatedInResultsDirectory(string runnerFramework, str
         }
 
         [CustomDataTestMethod]
-        [NETFullTargetFramework]
+        [NETFullTargetFramework(true, true)]
```
Comment (smadala) [inline test/Microsoft.TestPlatform.AcceptanceTests/ResultsDirectoryTests.cs]: No need to run in inprocess.

File: test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs [hunk 5]
```diff
@@ -9,68 +9,72 @@ namespace Microsoft.TestPlatform.AcceptanceTests
     public class TestCaseFilterTests : AcceptanceTestBase
     {
         [CustomDataTestMethod]
-        [NETFullTargetFramework]
+        [NETFullTargetFramework(true, true)]
```
Comment (smadala) [inline test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs]: Run only one test in each class in `inprocess`, Unless there is any behavior change.

File: test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs (hunk 2 above)
Comment (Faizan2304) [reply test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs]: Fixed

File: test/Microsoft.TestPlatform.AcceptanceTests/ResultsDirectoryTests.cs (hunk 3 above)
Comment (Faizan2304) [reply test/Microsoft.TestPlatform.AcceptanceTests/ResultsDirectoryTests.cs]: fixed

File: test/Microsoft.TestPlatform.AcceptanceTests/ResultsDirectoryTests.cs (hunk 4 above)
Comment (Faizan2304) [reply test/Microsoft.TestPlatform.AcceptanceTests/ResultsDirectoryTests.cs]: fixed

File: test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs (hunk 5 above)
Comment (Faizan2304) [reply test/Microsoft.TestPlatform.AcceptanceTests/TestCaseFilterTests.cs]: Fixed

**Review by smadala — APPROVED** [review]

**Pull request merged**